api/src/main/java/org/apache/iceberg/transforms/SortOrderVisitor.java

```diff
@@ -85,21 +85,18 @@ static <R> List<R> visit(SortOrder sortOrder, SortOrderVisitor<R> visitor) {
             visitor.truncate(
                 sourceName, field.sourceId(), width, field.direction(), field.nullOrder()));
       } else if (transform == Dates.YEAR
-          || transform == Timestamps.YEAR
+          || transform == Timestamps.YEAR_FROM_MICROS
+          || transform == Timestamps.YEAR_FROM_NANOS
           || transform instanceof Years) {
         results.add(
             visitor.year(sourceName, field.sourceId(), field.direction(), field.nullOrder()));
-      } else if (transform == Dates.MONTH
-          || transform == Timestamps.MONTH
-          || transform instanceof Months) {
+      } else if ("month".equalsIgnoreCase(transform.toString())) {
```

Comment:
is there a particular reason to switch to string comparison here and in other places?